plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -193,8 +196,10 @@ public void setFilterPath(File path) {
 
     @Override
     protected boolean doCheckState() {
+      String directory = getStringValue();
       if (!super.doCheckState()) {
-        status = new Status(IStatus.ERROR, getClass().getName(), "Invalid directory");
+        String message = MessageFormat.format(SdkUiMessages.NoSuchDirectory, directory);
+        status = new Status(IStatus.ERROR, getClass().getName(), message);
         return false;
       }
       status = Status.OK_STATUS;
```

Comment:
I think you can replace two instances of `getStringValue()` in the next line with the variable `directory`.